File: ios/Classes/model/SipConfiguration.swift
//
//  SipConfiguration.swift
//  voip24h_sdk_mobile
//
//  Created by Phát Nguyễn on 12/08/2022.
//

import Foundation
import linphonesw


class SipConfiguaration: Decodable {
    
    var ext: String!
    var password: String!
    var domain: String!
    var port: Int = 5060
    var transportType: String = ""
    var isKeepAlive: Bool = false
    
    
    func toLpTransportType() -> TransportType {
        switch(transportType) {
            case "Tcp":
                return TransportType.Tcp
            case "Ddp":
                return TransportType.Udp
            case "Tls":
                return TransportType.Tls
            default:
                return TransportType.Udp
        }
    }
    
    private enum CodingKeys : String, CodingKey {
        case ext = "extension", password, domain, port, transportType, isKeepAlive
    }
}

File: android/src/main/kotlin/voip24h/sdk/mobile/voip24h_sdk_mobile/model/SipConfiguration.kt
package voip24h.sdk.mobile.voip24h_sdk_mobile.model

import org.linphone.core.TransportType

data class SipConfiguration(
    val extension: String,
    val domain: String,
    val password: String,
    var port: Int = 5060,
    var transportType: String = TransportType.Udp.name,
    var isKeepAlive: Boolean = false
) {
    fun toLpTransportType() : TransportType = TransportType.valueOf(transportType)
}